samtranslator/model/api/api_generator.py

```diff
@@ -290,10 +290,11 @@ def _construct_api_domain(self, rest_api):
         if endpoint is None:
             endpoint = "REGIONAL"
             self.domain["EndpointConfiguration"] = "REGIONAL"
-        elif endpoint not in ["EDGE", "REGIONAL"]:
+        elif endpoint not in ["EDGE", "REGIONAL", "PRIVATE"]:
             raise InvalidResourceException(
                 self.logical_id,
-                "EndpointConfiguration for Custom Domains must be" " one of {}.".format(["EDGE", "REGIONAL"]),
+                "EndpointConfiguration for Custom Domains must be"
+                " one of {}.".format(["EDGE", "REGIONAL", "PRIVATE"]),
```

Comment:
@sriram-mv / @scbrown Custom Domains are not supported on private APIs, right?

- https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-custom-domains.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-domainname-endpointconfiguration.html